modules/apps/27-interchain-accounts/controller/keeper/events.go

```diff
@@ -13,22 +15,18 @@ import (
 
 // EmitAcknowledgementEvent emits an event signalling a successful or failed acknowledgement and including the error
 // details if any.
-func EmitAcknowledgementEvent(ctx context.Context, packet channeltypes.Packet, ack exported.Acknowledgement, err error) {
-	sdkCtx := sdk.UnwrapSDKContext(ctx) // TODO: https://github.com/cosmos/ibc-go/issues/5917
-	attributes := []sdk.Attribute{
-		sdk.NewAttribute(sdk.AttributeKeyModule, icatypes.ModuleName),
-		sdk.NewAttribute(icatypes.AttributeKeyControllerChannelID, packet.GetDestChannel()),
-		sdk.NewAttribute(icatypes.AttributeKeyAckSuccess, strconv.FormatBool(ack.Success())),
+func (k *Keeper) EmitAcknowledgementEvent(ctx context.Context, packet channeltypes.Packet, ack exported.Acknowledgement, err error) {
+	attributes := []event.Attribute{
+		event.NewAttribute(sdk.AttributeKeyModule, icatypes.ModuleName),
+		event.NewAttribute(icatypes.AttributeKeyControllerChannelID, packet.GetDestChannel()),
+		event.NewAttribute(icatypes.AttributeKeyAckSuccess, strconv.FormatBool(ack.Success())),
 	}
 
 	if err != nil {
-		attributes = append(attributes, sdk.NewAttribute(icatypes.AttributeKeyAckError, err.Error()))
+		attributes = append(attributes, event.NewAttribute(icatypes.AttributeKeyAckError, err.Error()))
 	}
 
-	sdkCtx.EventManager().EmitEvent(
-		sdk.NewEvent(
-			icatypes.EventTypePacket,
-			attributes...,
-		),
-	)
+	if err := k.EventService.EventManager(ctx).EmitKV(icatypes.EventTypePacket, attributes...); err != nil {
+		panic(err)
```

Comment:
must return error

Reply:
in what circumstances would this error? This would seem like a panic'able situation (i.e in what way would callee of this func handle this error?)

Reply:
the practical answer is - this will never error right now given the default implementation, but given that this is an interface and `EventService` may be swappable to a custom type in some future reality (e.g. like a kafka event service or something) then there may be an error surfaced. 

My take was to just propagate the errors as good practise and not just panic. 

We could just take the most lazy approach and panic in all cases at the moment, but its basically unreachable code rn.